File: lab 1/src/domain/book.py
class Book:
    def __init__(self, title, author, isbn, price):
        self.title = title
        self.author = author
        self.isbn = isbn
        self.price = price

    def __str__(self):
        return f"{self.title} by {self.author.name} (ISBN: {self.isbn})"

File: lab2/src/domain/Book.ts
import { Author } from "./Author";

export class Book {
  constructor(
    public title: string,
    public author: Author,
    public year: number,
    public id?: number
  ) {}

  getDetails(): string {
    return `${this.title} by ${this.author.name} (${this.year})`;
  }
}
